packages/ckeditor5-engine/src/controller/editingcontroller.ts

```diff
@@ -111,7 +111,11 @@ export default class EditingController extends ObservableMixin() {
 		// Whenever model document is changed, convert those changes to the view (using model.Document#differ).
 		// Do it on 'low' priority, so changes are converted after other listeners did their job.
 		// Also convert model selection.
-		this.listenTo<DocumentChangeEvent>( doc, 'change', () => {
+		this.listenTo<DocumentChangeEvent>( doc, 'change', evt => {
+			if ( evt.name === 'change:isReadOnly' ) {
+				return;
+			}
```

Comment:
It's a major problem to have a `change` event fired by a class "naturally", i.e. the class is not an observable. It's painful and we will lose the symmetry with the view but `change` event is too widely spread to have it this way. So, we cannot make `model.Document` an observable. Please propose a different solution.